### ISSUE TITLE
Different UI elements for steps not working

### ISSUE THREAD

**Issue by CollinAlpert** [badrinathvm/StepperView#75]
**Describe the bug**
Using different UI elements for different steps does not seem to work.

**To Reproduce**
```swift
    static let stepLineColor = Color(red: 229 / 255, green: 229 / 255, blue: 229 / 255)
    static let stepCircleColor = Color(red: 17 / 255, green: 16 / 255, blue: 56 / 255)

    let indicationTypes: [StepperIndicationType<NumberedCircleView>] = [
        .custom(NumberedCircleView(text: "1", color: stepCircleColor, triggerAnimation: true)),
        .custom(NumberedCircleView(text: "2", color: stepCircleColor, triggerAnimation: true)),
        .custom(NumberedCircleView(text: "3", color: stepCircleColor, triggerAnimation: true)),
        .custom(NumberedCircleView(text: "4", color: stepCircleColor, triggerAnimation: true)),
        .custom(NumberedCircleView(text: "5", color: stepCircleColor, triggerAnimation: true)),
        .custom(NumberedCircleView(text: "6", color: stepCircleColor, triggerAnimation: true))
    ]


    var body: some View {
        StepperView()
                .addSteps([
                    VStack {
                        Text("")
                        DatePicker("", selection: $date, in: Date()..., displayedComponents: [.date])
                    },
                    Text("")
                ])
                .indicators(indicationTypes)
                .stepIndicatorMode(StepperMode.vertical)
                .spacing(30)
                .lineOptions(StepperLineOptions.custom(2, TestView.stepLineColor))
    }
```

This yields in `Failed to produce diagnostic for expression; please submit a bug report (https://swift.org/contributing/#reporting-bugs) and include the project`

**Expected behavior**
The code compiles, the first step displays a `VStack` and the second step displays Text.

**Screenshots**

**Info (please complete the following information):**
 - Device: Simulator
 - OS: 14.5
 - SPM library version: 1.6.2
 - Xcode Version: 12.5

**Comment by badrinathvm** [comment]
Hi @CollinAlpert  Thank you for checking out StepperView. I am looking over the issue and update u shortly.

**Comment by badrinathvm** [comment]
Hi @CollinAlpert  in `addSteps`, each view has to be of type `AnyView` in simple u can use `eraseToAnyView()` view modifier. See the below code.  Also `number of Steps === number of Indicators.`

```
@available(iOS 13.0, OSX 10.15, tvOS 13.0, watchOS 6.0, *)
struct CheckTestView: View {
    @State private var date = Date()
    static let stepLineColor = Color(red: 229 / 255, green: 229 / 255, blue: 229 / 255)
    static let stepCircleColor = Color(red: 17 / 255, green: 16 / 255, blue: 56 / 255)
    
    let indicationTypes: [StepperIndicationType<NumberedCircleView>] = [
        .custom(NumberedCircleView(text: "1", color: stepCircleColor, triggerAnimation: true)),
        .custom(NumberedCircleView(text: "2", color: stepCircleColor, triggerAnimation: true))
    ]
    
    var body: some View {
            StepperView()
                .addSteps([
                    VStack {
                        Text("").eraseToAnyView()
                        DatePicker("", selection: $date, in: Date()..., displayedComponents: [.date])
                    }.eraseToAnyView(),
                    Text("").eraseToAnyView()
                ])
                .indicators(indicationTypes)
                .stepIndicatorMode(StepperMode.vertical)
                .spacing(30)
                .lineOptions(StepperLineOptions.custom(2, CheckTestView.stepLineColor))
    }
}
```
Let me know if any more details required.

**Comment by CollinAlpert** [comment]
Thanks, this resolved my issue.
